xmtp_mls/src/storage/encrypted_store/db_connection.rs

```diff
@@ -12,26 +12,37 @@ enum RefOrValue<'a, T> {
 /// and transaction state can be shared between the OpenMLS Provider and
 /// native XMTP operations
 pub struct DbConnection<'a> {
-    wrapped_conn: RefCell<RefOrValue<'a, RawDbConnection>>,
+    wrapped_conn: Mutex<RefOrValue<'a, RawDbConnection>>,
 }
 
 impl<'a> DbConnection<'a> {
     pub(crate) fn new(conn: &'a mut RawDbConnection) -> Self {
         Self {
-            wrapped_conn: RefCell::new(RefOrValue::Ref(conn)),
+            wrapped_conn: Mutex::new(RefOrValue::Ref(conn)),
```

Comment:
Generally curious about this change

Reply:
Updated the README with this detail:
We use Tokio as our [async runtime](https://rust-lang.github.io/async-book/08_ecosystem/00_chapter.html). Uniffi can use this runtime on async methods and objects using the annotation `#[uniffi::export(async_runtime = ‘tokio’)]`. Uniffi plumbs up an executor (scheduler) in the foreign language to the Tokio runtime in Rust. More details [here](https://github.com/mozilla/uniffi-rs/blob/734050dbf1493ca92963f29bd3df49bb92bf7fb2/uniffi_core/src/ffi/rustfuture.rs#L11-L18). By default, Tokio's [multi-thread scheduler](https://docs.rs/tokio/latest/tokio/runtime/index.html#multi-thread-scheduler) is used. This means that functions may resume execution in a different thread when an `await` is encountered. This in turn means that structs held across an `await` in libxmtp must be `Send` and `Sync`.

We could also consider switching to the single-thread executor, if uniffi allows us to specify it - I haven't investigated this yet.